honeycomb-core/src/cells/attributes.rs

```diff
@@ -0,0 +1,107 @@
+//! Generic attributes implementation
+//!
+//! This module contains all code used to handle attribute genericity in the context of mesh
+//! representation through combinatorial maps embedded data.
+
+// ------ IMPORTS
+
+use crate::OrbitPolicy;
+
+// ------ CONTENT
+
+/// Generic attribute trait for logical behavior
+///
+/// This trait can be implemented for a given attribute in order to define the behavior to
+/// follow when (un)sewing operations result in an update of the attribute.
+///
+/// # Example
+///
+/// For an intensive property of a system (e.g. a temperature), a dummy implementation would look
+/// like this:
+///
+/// ```rust
+/// use honeycomb_core::AttributeLogic;
+///
+/// #[derive(Copy, Clone)]
+/// pub struct Temperature {
+///     pub val: f32
+/// }
+///
+/// impl AttributeLogic for Temperature {
+///     fn merge(lhs: Self, rhs: Self) -> Self {
+///         Temperature { val: (lhs.val + rhs.val) / 2.0 }
+///     }
+///
+///     fn split(lhs: Self) -> (Self, Self) {
+///         (lhs, lhs)
+///     }
+///
+///     fn merge_undefined(lhs: Option<Self>) -> Self {
+///         lhs.unwrap_or(Temperature { val: 0.0 })
+///     }
+/// }
+/// ```
+pub trait AttributeLogic: Sized {
```

Comment:
`AttributeLogic` is not very appealing. Can something like `AttributeMergeable` or `AttributeMergeTrait` or `AttributeMerge` be easier to understand.

So next question, is there a reason to tie `merge` and `split` together or can we have algorithm that can deal with only `merge` or only `split`?

Reply:
From what I understood from my discussion with @franck-ledoux, `merge` and `split` are always defined for a given attribute. Also, from a user API perspective, I think keeping the number of trait to a minimum is the best choice for both clarity & simplicity.